src/clp_ffi_py/ir/native/PyDeserializer.cpp

```diff
@@ -279,7 +289,12 @@ auto PyDeserializer::handle_log_event(clp::ffi::KeyValuePairLogEvent&& log_event
         // deserialized log event.
         clear_deserialized_log_event();
     }
-    m_deserialized_log_event = new clp::ffi::KeyValuePairLogEvent{std::move(log_event)};
+    m_deserialized_log_event
+            = new (std::nothrow) clp::ffi::KeyValuePairLogEvent{std::move(log_event)};
+    if (nullptr == m_deserialized_log_event) {
+        // TODO: Set this to a proper error code when user-defined error code is supported.
+        return IRErrorCode::IRErrorCode_Eof;
```

Comment:
maybe we should return IRErrorCode::NoMem. 
Unfortuately it will require updating clp repo so obviously out of scope of this PR, lol

Reply:
Sigh, correct. I'm trying to proceed error handling improvements. Hopefully we will have the user-defined user code supported soon.